File: zad1/ClientApp/src/models/mutation/SwapMutation.ts
import IMutation from "./IMutation";
import Individual from "../Individual";
import _ from "lodash";

export default class SwapMutation implements IMutation {
    evaluate(individual: Individual): Individual {
        let firstIndex = ~~(Math.random() * individual.towns.length);
        let secondIndex = ~~(Math.random() * individual.towns.length);

        const clone = _.cloneDeep(individual);

        const temp = clone.towns[firstIndex];
        clone.towns[firstIndex] = clone.towns[secondIndex];
        clone.towns[secondIndex] = temp;

        return clone;
    }
}

File: zad1/Algorithms/Mutation/SwapMutation.cs
using System;
using zad1.Models;

namespace zad1.Algorithms.Mutation
{
    public class SwapMutation : IMutation
    {
        public Individual Evaluate(Individual individual)
        {
            var rng = new Random();

            var firstIndex = rng.Next(individual.Towns.Count);
            var secondIndex = rng.Next(individual.Towns.Count);

            var clone = (Individual) individual.Clone();

            var temp = clone.Towns[firstIndex];
            clone.Towns[firstIndex] = clone.Towns[secondIndex];
            clone.Towns[secondIndex] = temp;

            return clone;
        }
    }
}